### PR TITLE
Feature84

### PR DESCRIPTION
**Pull request by AgnieszkaKurek**

### CODE REVIEW

File: functions/exercise26/script.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/// <reference path="../../typings/globals/jasmine/index.d.ts" />
+/*globals getLongestSubstringWithoutRepeatingCharacters*/
+
+describe("Find longest substring in a given a string without longest substring", (() => {
```
Comment (Krzysiek102) [inline functions/exercise26/script.js]: I do not understand this description.

File: functions/exercise26/script.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    
+        it("The longest substring in a given a string without longest substring is 'lphant' ", (() => {
```
Comment (Krzysiek102) [inline functions/exercise26/script.js]: I do not understand this description.

File: functions/exercise26/script.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+            let substringWithoutRepeatingCharacters = getLongestSubstringWithoutRepeatingCharacters(["gorilla", "gitaffe", "parrot", "elephant"]);
```
Comment (Krzysiek102) [inline functions/exercise26/script.js]: Incorrect type of input parameter.

**Review by Krzysiek102 — CHANGES_REQUESTED** [review]
I think you misunderstood the requirements. Please read it once again.

File: functions/exercise26/scriptSpec.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+/// <reference path="../../typings/globals/jasmine/index.d.ts" />
+/*globals getLongestSubstringWithoutRepeatingCharacters*/
+
+describe(" Find longest substring in a given a string without repeating characters", (() => {
+    
+         it("The longest substring in a given a string without repeating characters is 'lphant' ", (() => {
```
Comment (Krzysiek102) [inline functions/exercise26/scriptSpec.js]: Incorrect description. Does it give any value to the user that reads it?

File: functions/exercise26/scriptSpec.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+             let substringWithoutRepeatingCharacters = getLongestSubstringWithoutRepeatingCharacters("Gorilla, gitaffe, parrot and  elephant");
```
Comment (Krzysiek102) [inline functions/exercise26/scriptSpec.js]: Can you put more simpler test here?

File: functions/exercise26/scriptSpec.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+             expect(substringWithoutRepeatingCharacters).toEqual("lphant");
```
Comment (Krzysiek102) [inline functions/exercise26/scriptSpec.js]: Incorrect result - there is no substring lphant in the string you specified.

**Review by Krzysiek102 — CHANGES_REQUESTED** [review]

File: functions/exercise26/script.js [hunk 7]
```diff
@@ -0,0 +1,3 @@
+
```
Comment (Krzysiek102) [inline functions/exercise26/script.js]: Please do not commit this file for noe

Comment (Krzysiek102) [reply functions/exercise26/script.js]: For now

File: functions/exercise26/scriptSpec.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+/// <reference path="../../typings/globals/jasmine/index.d.ts" />
+/*globals getLongestSubstringWithoutRepeatingCharacters*/
+
+describe(" Find longest substring in a given a string without repeating characters", (() => {
+    
+         it("If the string consists of one word 'elephant', then the longest substring without repeating characters is 'lphant' ", (() => {
```
Comment (Krzysiek102) [inline functions/exercise26/scriptSpec.js]: Remove space at the end

**Review by Krzysiek102 — CHANGES_REQUESTED** [review]

File: functions/exercise26/scriptSpec.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+/// <reference path="../../typings/globals/jasmine/index.d.ts" />
+/*globals getLongestSubstringWithoutRepeatingCharacters*/
+
+describe(" Find longest substring in a given a string without repeating characters", (() => {
+    
+         it("If the string consists of one word 'elephant', then the longest substring without repeating characters is 'lphant'", (() => {
+             let substringWithoutRepeatingCharacters = getLongestSubstringWithoutRepeatingCharacters("elephant");
+             expect(substringWithoutRepeatingCharacters).toEqual("lphant");
```
Comment (Krzysiek102) [inline functions/exercise26/scriptSpec.js]: It is not correct, the longest is phant. There is no such substring lphant as l occurs before e that repeats

**Review by Krzysiek102 — CHANGES_REQUESTED** [review]

File: functions/exercise26/scriptSpec.js (hunk 9 above)
Comment (AgnieszkaKurek) [reply functions/exercise26/scriptSpec.js]: L do not repeat. Example from the exercise: "google.com"; 
"example.com";                                       Result:"gle.com","xample.co"

Comment (AgnieszkaKurek) [reply functions/exercise26/scriptSpec.js]: That is, in this exercise, the point is that the letters that are repeated are only once

Comment (AgnieszkaKurek) [reply functions/exercise26/scriptSpec.js]: the result should be an lephant

File: functions/exercise26/scriptSpec.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+             let substringWithoutRepeatingCharacters = getLongestSubstringWithoutRepeatingCharacters("elephant");
+             expect(substringWithoutRepeatingCharacters).toEqual(["lephant"]);
+         }));
+         
+         it("If the string consists of one word 'abba', then the longest substrings without repeating characters are 'ab' 'ba'", (() => {
+            let substringWithoutRepeatingCharacters = getLongestSubstringWithoutRepeatingCharacters("abba");
+            expect(substringWithoutRepeatingCharacters).toEqual(["ab","ba"]);
+        }));
+    
+
```
Comment (Krzysiek102) [inline functions/exercise26/scriptSpec.js]: No need for this empty line

File: functions/exercise26/scriptSpec.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
+/// <reference path="../../typings/globals/jasmine/index.d.ts" />
+/*globals getLongestSubstringWithoutRepeatingCharacters*/
```
Comment (Krzysiek102) [inline functions/exercise26/scriptSpec.js]: Substring or substrings?

**Review by Krzysiek102 — CHANGES_REQUESTED** [review]

File: functions/exercise26/scriptSpec.js [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+
+describe(" Find longest substring in a given a string without repeating characters", (() => {
+
+    it("If the string consists of one word 'elephant', then the longest substring without repeating characters is 'lephant'", (() => {
+        let substringsWithoutRepeatingCharacters = getLongestSubstringsWithoutRepeatingCharacters("elephant");
+        expect(substringsWithoutRepeatingCharacters).toEqual(["lephant"]);
+    }));
+
+    it("If the string consists of one word 'abba', then the longest substrings without repeating characters are 'ab' 'ba'", (() => {
+        let substringWithoutRepeatingCharacters = getLongestSubstringsWithoutRepeatingCharacters("abba");
```
Comment (Krzysiek102) [inline functions/exercise26/scriptSpec.js]: substring or substrings?

**Review by Krzysiek102 — CHANGES_REQUESTED** [review]

**Review by Krzysiek102 — APPROVED** [review]

**Pull request merged**